cryptographic_estimators/MAYOEstimator/MAYOAlgorithms/intersection_attack.py

```diff
@@ -0,0 +1,104 @@
+# ****************************************************************************
+# Copyright 2023 Technology Innovation Institute
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# ****************************************************************************
+
+
+from ..mayo_algorithm import MAYOAlgorithm
+from ..mayo_problem import MAYOProblem
+from cryptographic_estimators.MQEstimator.MQAlgorithms.booleansolve_fxl import BooleanSolveFXL
+from cryptographic_estimators.MQEstimator.mq_problem import MQProblem
+from math import log2
+
+
+class IntersectionAttack(MAYOAlgorithm):
+    """
+    Construct an instance of IntersectionAttack estimator
+
+    The intersection attack [Beu20]_ generalizes the ideas behind the Kipnis-Shamir attack, in
+    combination with a system-solving approach such as in the reconciliation attack.
+
+    INPUT:
+
+    - ``problem`` -- DummyProblem object including all necessary parameters
+    - ``w`` -- linear algebra constant (default: Obtained from MAYOAlgorithm)
+    - ``h`` -- external hybridization parameter (default: 0)
+    - ``nsolutions`` -- number of solutions in logarithmic scale (default: expected_number_solutions))
+    - ``excluded_algorithms`` -- a list/tuple of MQ algorithms to be excluded (default: [Lokshtanov])
+    - ``memory_access`` -- specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage)
+    - ``complexity_type`` -- complexity type to consider (0: estimate, default: 0)
+    - ``bit_complexities`` -- determines if complexity is given in bit operations or basic operations (default 1: in bit)
+
+    """
+
+    def __init__(self, problem: MAYOProblem, **kwargs):
+        super().__init__(problem, **kwargs)
+
+        self._name = "IntersectionAttack"
+        self._attack_type = "key-recovery"
```

Comment:
this string should be a constant in the base constant file and then imported (as it has to be used in this exact form everywhere)

Reply:
I agree.